### PR TITLE
JNI support with CI workflow

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview

This PR adds JNI support to compilation of native code for https://github.com/bitcoin-core/secp256k1.

JNI files are taken from PR which removed support on the source repo.
https://github.com/bitcoin-core/secp256k1/pull/682

Also files are updated from the PR on https://github.com/bitcoin-s/bitcoin-s repo which also uses original source.
https://github.com/bitcoin-s/secp256k1-zkp/pull/1/commits/17ab106e0eaaa3c9607bd748328930b551d05ec4 (PR https://github.com/bitcoin-s/secp256k1-zkp/pull/1)

Also fix for macOS for JDK path is taken from @nzpr's build.
https://github.com/nzpr/secp256k1/commit/41fa378430c8164fc968a2f48f5201bc30d46065

### GitHub CI workflow

GitHub workflow CI files are added with support for bors. Running examples can be seen on the fork repo.
https://github.com/tgrospic/secp256k1-native/actions/runs/2851923440

## TODO

_Makefile and configure are for now copied as a whole files, but later only changes should be extracted to a script of diff file to update originals._

What's missing is support for multiple architecture e.g. ARM and CI action to create releases.

### CODE REVIEW

**Comment by tgrospic** [comment]
bors try

File: jni/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,241 @@
+.PHONY: clean-precomp precomp
```
Comment (nzpr) [inline jni/Makefile.am:1]: Would it make sense to have a patch for original Makefile here instead of rewriting the whole file? Same for `configure.ac`

Comment (tgrospic) [reply jni/Makefile.am:1]: This is an initial version with the copy of the whole files. Changes should be extracted to a script or diff file.

**Review by nzpr — APPROVED** [review]

**Review by nzpr — APPROVED** [review]

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**